zulipterminal/ui_tools/boxes.py

```diff
@@ -445,11 +445,23 @@ def autocomplete_users(self, text: str, prefix_string: str
                                        reverse=True)
 
         user_names = [user['full_name'] for user in sorted_matching_users]
+
+        user_name_duplicates = [user_name
+                                for count, user_name in enumerate(user_names)
+                                if user_name in user_names[:count]]
```

Comment:
This remains my main concern with this implementation. It's not a common case, but for example I get a pretty bad response trying a mention of `@user`<kbd>ctrl-f</kbd> on czo.

Instinctively I'd suggest using `Counter`, but anything which doesn't make a slice in every loop would likely be sufficient.

You have tests to ensure you get good results, so refactoring should be easier :+1: